dashboard/src/components/applications/AppDeployments.tsx

```diff
@@ -226,42 +79,61 @@ export default function AppDeployments(props: AppDeploymentsProps) {
     },
   });
 
-  useEffect(() => {
-    if (!data) {
-      return;
-    }
-
-    if (page === 1) {
-      setIdOfLiveDeployment(getLastLiveDeployment(data.deployments));
-    }
-  }, [data, idOfLiveDeployment, loading, page]);
+  const {
+    data: scheduledOrPendingDeploymentsData,
+    loading: scheduledOrPendingDeploymentsLoading,
+  } = useScheduledOrPendingDeploymentsSubSubscription({
+    variables: {
+      appId,
+    },
+  });
 
-  if (loading) {
-    return <DelayedLoading delay={500} className="mt-12" />;
+  if (loading || scheduledOrPendingDeploymentsLoading) {
+    return (
+      <ActivityIndicator
+        delay={500}
+        className="mt-12"
+        label="Loading deployments..."
+      />
+    );
   }
 
   if (error) {
     throw error;
   }
 
-  const nrOfDeployments = data.deployments.length;
+  const { deployments } = data || {};
+  const { deployments: scheduledOrPendingDeployments } =
+    scheduledOrPendingDeploymentsData || {};
+
+  const nrOfDeployments = deployments?.length || 0;
   const nextAllowed = !(nrOfDeployments < limit);
+  const liveDeploymentId = getLastLiveDeployment(deployments);
 
   return (
     <div className="mt-6">
       {nrOfDeployments === 0 ? (
         <p className="text-sm text-greyscaleGrey">No deployments yet.</p>
       ) : (
         <div>
-          <div className="mt-3 divide-y-1 border-t border-b">
-            {data.deployments.map((deployment) => (
-              <AppDeploymentRow
-                deployment={deployment}
+          <List className="mt-3 divide-y-1 border-t border-b">
+            {deployments.map((deployment, index) => (
+              <DeploymentListItem
                 key={deployment.id}
-                isDeploymentLive={idOfLiveDeployment === deployment.id}
+                deployment={deployment}
+                isLive={liveDeploymentId === deployment.id}
```

Comment:
Thanks for merging these components together. But should we even show the redeploy button here? I think it looks a bit too crowded:  
  
 

![](https://user-images.githubusercontent.com/20285232/211797601-6e4284b0-3b8b-4c95-83f8-1ceec9cf8325.png)